src/main/java/uk/gov/ons/census/casesvc/service/UacService.java

```diff
@@ -53,16 +54,18 @@ public UacService(
     this.caseService = caseService;
   }
 
-  public UacQidLink buildUacQidLink(Case caze, int questionnaireType) {
-    return buildUacQidLink(caze, questionnaireType, null);
-  }
-
-  public UacQidLink buildUacQidLink(Case caze, int questionnaireType, UUID batchId) {
+  public UacQidLink buildUacQidLink(
+      Case caze, int questionnaireType, UUID batchId, EventDTO sourceEvent) {
     UacQidDTO uacQid = uacQidCache.getUacQidPair(questionnaireType);
-    return buildUacQidLink(caze, batchId, uacQid.getUac(), uacQid.getQid());
+    return buildUacQidLink(caze, batchId, sourceEvent, uacQid.getUac(), uacQid.getQid());
   }
 
-  private UacQidLink buildUacQidLink(Case linkedCase, UUID batchId, String uac, String qid) {
+  private UacQidLink buildUacQidLink(
+      Case linkedCase, UUID batchId, EventDTO sourceEvent, String uac, String qid) {
+    UacQidLinkMetadata metadata = new UacQidLinkMetadata();
```

Comment:
this object is created but not persisted anywhere?  the tests still pass whether or not these 3 lines exist?

Reply:
fixed